### PR TITLE
Implemented the serial synchronous and interrupt functions

### PR DESCRIPTION
**Pull request by wajahat-abbas**
Notes:
* Pull requests will not be accepted until the submitter has agreed to the [contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md).
* This is just a template, so feel free to use/remove the unnecessary things

## Description
A few sentences describing the overall goals of the pull request's commits.

## Status
**READY/IN DEVELOPMENT/HOLD**

## Migrations
If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

YES | NO

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos
- [ ] Tests
- [ ] Documentation

## Deploy notes
Notes regarding the deployment of this PR. These should note any
required changes in the build environment, tools, compilers, etc.

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by wajahat-abbas** [comment]
@fahim-ublox Please review

**Comment by wajahat-abbas** [comment]
[Jenkins build](http://jenkins.pk-lhr.u-blox.net:8080/job/mbed/job/KM/job/Pull%20request%20build/25/)

File: targets/TARGET_ublox/TARGET_R5030/serial_api.c [hunk 1]
```diff
@@ -151,71 +171,107 @@ uint32_t zeroBitNumFromLsb(uint32_t data)
...
 {
-    // if (obj->tx_pin == UART0_TX) {
-        // stdio_uart_inited = 0;
-    // }
-
-    // serial_irq_ids[obj->index] = 0;
-
-    // /* Release the port HW */
-    // deinit_config(obj);
+    if (obj->tx_pin == UART1_TX) {
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/serial_api.c:262]: Do we need check on UART2_TX ?

Comment (wajahat-abbas) [reply targets/TARGET_ublox/TARGET_R5030/serial_api.c:262]: Its not needed as stdio_uart_inited is set only for UART1.

Comment (fahimalavi) [reply targets/TARGET_ublox/TARGET_R5030/serial_api.c:262]: make sense

**Review by fahimalavi — APPROVED** [review]

**Review by RobMeades — COMMENTED** [review]
Great to see the implementation coming on; no particular comments from me, you know more than I do on the detail here.

**Pull request merged**